Tasks/AzureRmWebAppDeploymentV4/operations/AzureAppServiceUtility.ts

```diff
@@ -4,30 +4,45 @@ import webClient = require('azure-arm-rest-v2/webClient');
 var parseString = require('xml2js').parseString;
 import Q = require('q');
 import { Kudu } from 'azure-arm-rest-v2/azure-arm-app-service-kudu';
+import { Package } from 'webdeployment-common-v2/packageUtility';
 
 export class AzureAppServiceUtility {
     private _appService: AzureAppService;
     constructor(appService: AzureAppService) {
         this._appService = appService;
     }
 
-    public async updateScmTypeAndConfigurationDetails() : Promise<void>{
+    public async updateScmTypeAndConfigurationDetails(packageArtifactAlias: string = null) : Promise<void>{
             try {
             var configDetails = await this._appService.getConfiguration();
             var scmType: string = configDetails.properties.scmType;
+            let shouldUpdateMetadata = false;
             if (scmType && scmType.toLowerCase() === "none") {
                 configDetails.properties.scmType = 'VSTSRM';
                 tl.debug('updating SCM Type to VSTS-RM');
                 await this._appService.updateConfiguration(configDetails);
                 tl.debug('updated SCM Type to VSTS-RM');
-                tl.debug('Updating metadata with latest release details');
-                await this._appService.patchMetadata(this._getNewMetadata());
-                tl.debug('Updated metadata with latest release details');
-                console.log(tl.loc("SuccessfullyUpdatedAzureRMWebAppConfigDetails"));
+                shouldUpdateMetadata = true;
+            }
+            else if(scmType && scmType.toLowerCase() == "vstsrm") {
+                tl.debug("SCM Type is VSTSRM");
+                shouldUpdateMetadata = true;
             }
             else {
                 tl.debug(`Skipped updating the SCM value. Value: ${scmType}`);
             }
+            
+            if(shouldUpdateMetadata) {
+                tl.debug('Updating metadata with latest pipeline details');
+                let newMetadataProperties = this._getNewMetadata(packageArtifactAlias);
+                let siteMetadata = await this._appService.getMetadata();
+                for(let property in newMetadataProperties) {
+                    siteMetadata.properties[property] = newMetadataProperties[property];
```

Comment:
You should also check that you are indeed updating some property. If the old metadata is same as current one - we should skip updating it.

Reply:
okay will do.